lib/chef/resource/timezone.rb

```diff
@@ -48,90 +56,118 @@ class Timezone < Chef::Resource
         description: "An optional property to set the timezone value if it differs from the resource block's name.",
         name_property: true
 
-      action :set do
-        description "Set the timezone."
-
-        # some linux systems may be missing the timezone data
-        if linux?
-          package "tzdata" do
-            package_name suse? ? "timezone" : "tzdata"
-          end
+      # detect the current TZ on darwin hosts
+      #
+      # @since 14.7
+      # @return [String] TZ database value
+      def current_darwin_tz
```

Comment:
Nitpick: should this be `curent_macos_tz`?

Reply:
Fair enough. I fixed the name